### PR TITLE
Import the Lookbook crate for component previews

### PR DESCRIPTION
**Pull request by matthunz**
Imports the [Lookbook](https://github.com/matthunz/lookbook) crate as a standard component preview system for Dioxus.

Screenshot 2024-08-07 at 2 36 27 AM

### Release Checklist
 - [x] Update to Dioxus v0.6 Alpha
 - [x] Add Dioxus branding and colors
 - [x] Import to the main Dioxus repo
 - [x] Update `dioxus-markdown` to Dioxus v0.6 https://github.com/DioxusLabs/markdown/pull/9
    ```
     #[no_mangle]
     fn preview_x() 
     ```
     - Search for functions in o files that has `preview` prefix
   
 I'd also love to import any of this crate's dependencies, such as:
- [`dioxus-material`](https://github.com/matthunz/dioxus-material)
- [`dioxus-spring`](https://github.com/matthunz/dioxus-spring)
- [`dioxus-resize-observer`](https://github.com/dioxus-community/dioxus-resize-observer)

In follow-up PRs I'd also like to:
 - Update `lookbook-macros` `dioxus-rsx` to `v0.6`
 - Integrate with the new `dx` linker system:

### CODE REVIEW

File: packages/lookbook/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+<div align="center">
+<h1>Lookbook</h1>
+<h4>UI preview framework for Dioxus<h4>
+<a href="https://crates.io/crates/lookbook">
+    <img src="https://img.shields.io/crates/v/lookbook?style=flat-square"alt="Crates.io version" />
+</a>
+<a href="https://docs.rs/lookbook/latest/lookbook">
+    <img src="https://img.shields.io/badge/docs-stable-blue.svg?style=flat-square"alt="docs.rs docs" />
+</a>
+<a href="https://dioxus-material.netlify.app/dioxus_material/">
```
Comment (marc2332) [inline packages/lookbook/README.md:10]: material? :trollface:

File: packages/lookbook/README.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    <img src="https://img.shields.io/badge/docs-nightly-blue.svg?style=flat-square"
+  alt="nightly docs" />
+</a>
+<a href="https://github.com/matthunz/dioxus-material/actions">
```
Comment (marc2332) [inline packages/lookbook/README.md:14]: same

File: packages/lookbook/README.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    <img src="https://github.com/matthunz/dioxus-material/actions/workflows/ci.yml/badge.svg"
```
Comment (marc2332) [inline packages/lookbook/README.md:15]: same

File: packages/lookbook/README.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  alt="CI status" />
+</div>
+<div align="center">
+    <a href="https://dioxus-material-lookbook.netlify.app/">Demo</a>
```
Comment (marc2332) [inline packages/lookbook/README.md:19]: You might want to change this to another domain eventually

File: packages/lookbook/README.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+fn main() {
+    dioxus::launch(app)
+}
+```
+
+## Usage
+First add Lookbook as a dependency to your project.
+
+```sh
+cargo add lookbook --git https://github.com/matthunz/lookbook
```
Comment (marc2332) [inline packages/lookbook/README.md:66]: Same

File: packages/lookbook/src/ui/pane.rs [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            }
+            {right}
+        }
+    )
+}
+
+#[component]
+pub fn VerticalPane(top: Element, bottom: Element) -> Element {
+    let container_ref = use_mounted();
+    let container_size = use_size(container_ref);
```
Comment (marc2332) [inline packages/lookbook/src/ui/pane.rs:42]: Why not use `use_resize`? This way you don't have to use `use_reactive`

File: packages/lookbook/Cargo.toml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+[package]
+name = "lookbook"
+version = "0.2.0-alpha.1"
+edition = "2021"
+license = "MIT OR Apache-2.0"
+
```
Comment (marc2332) [inline packages/lookbook/Cargo.toml:6]: Maybe add the rest of the metadata.

File: packages/lookbook/Cargo.toml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+[dependencies]
+dioxus = "0.6.0-alpha.1"
+dioxus-router = "0.6.0-alpha.0"
+dioxus-material = "0.3.0-alpha.1"
+dioxus-use-mounted = "0.3.0-alpha.1"
+dioxus-resize-observer = "0.3.0-alpha.1"
+dioxus_markdown = { git = "https://github.com/matthunz/markdown", branch = "v0.6" }
+lookbook-macros = { path = "../lookbook-macros" }
```
Comment (marc2332) [inline packages/lookbook/Cargo.toml:14]:
```suggestion
dioxus = { workspace = true }
dioxus-router = { workspace = true }
dioxus-material = "0.3.0-alpha.1"
dioxus-use-mounted = "0.3.0-alpha.1"
dioxus-resize-observer = "0.3.0-alpha.1"
dioxus_markdown = { git = "https://github.com/matthunz/markdown", branch = "v0.6" }
lookbook-macros = { workspace = true }
```

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: packages/lookbook/README.md (hunk 1 above)
Comment (matthunz) [reply packages/lookbook/README.md]: 😅 thanks

File: packages/lookbook/README.md (hunk 4 above)
Comment (matthunz) [reply packages/lookbook/README.md:15]: Good point 👍 I'm wondering what we should do with this...

Maybe it makes more sense to just show a section in the docs with a demo/screenshot?

Comment (marc2332) [reply packages/lookbook/README.md:15]: Maybe a gif

**Review by jkelleyrtp — CHANGES_REQUESTED** [review]
Sorry this has sat for so long...

Lookbook can't depend on any dioxus crates that are out of tree.

Also, if we were to merge lookbook support we need support within the CLI or other developer tooling. This PR doesn't add any integrations with the rest of the workspace so it wouldn't make much sense to add this in lieu of lookbook being out-of-tree.

I'm not sure what that integration would look like per se, but something like `dx preview` and automatic lookbook generation with the linker would be ideal.

Plus, this doesn't adjust any imports or add any feature flags to the `dioxus` crate, so users would need to import lookbook as a crate on their own.

**Pull request closed without merging**